### ISSUE TITLE
Why not log output?

### ISSUE THREAD

**Issue by jinfagang** [open-mmlab/mmdetection#2452]
Training solov2, I got nothing output:

```
# model settings
model = dict(
    type='SOLO',
    pretrained='torchvision://resnet50',
    backbone=dict(
        type='ResNet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3), # C2, C3, C4, C5
        frozen_stages=1,
        style='pytorch'),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        start_level=0,
        num_outs=5),
    bbox_head=dict(
        type='SOLOV2Head',
        num_classes=81,
        in_channels=256,
        stacked_convs=4,
        seg_feat_channels=256,
        strides=[8, 8, 16, 32, 32],
        scale_ranges=((1, 96), (48, 192), (96, 384), (192, 768), (384, 2048)),
        sigma=0.2,
        num_grids=[40, 36, 24, 16, 12],
        cate_down_pos=0,
        with_deform=False,
        loss_ins=dict(
            type='DiceLoss',
            use_sigmoid=True,
            loss_weight=3.0),
        loss_cate=dict(
            type='FocalLoss',
            use_sigmoid=True,
            gamma=2.0,
            alpha=0.25,
            loss_weight=1.0),
    ))
# training and testing settings
train_cfg = dict()
test_cfg = dict(
    nms_pre=500,
    score_thr=0.1,
    mask_thr=0.5,
    update_thr=0.05,
    kernel='gaussian',  # gaussian/linear
    sigma=2.0,
    max_per_img=100)
# dataset settings
dataset_type = 'CocoDataset'
data_root = 'data/coco/'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True, with_mask=True),
    dict(type='Resize',
         img_scale=[(1333, 800), (1333, 768), (1333, 736),
                    (1333, 704), (1333, 672), (1333, 640)],
         multiscale_mode='value',
         keep_ratio=True),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels', 'gt_masks']),
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(1333, 800),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(type='Normalize', **img_norm_cfg),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img']),
        ])
]
data = dict(
    imgs_per_gpu=2,
    workers_per_gpu=4,
    train=dict(
        type=dataset_type,
        ann_file=data_root + 'annotations/instances_train2017.json',
        img_prefix=data_root + 'train2017/',
        pipeline=train_pipeline),
    val=dict(
        type=dataset_type,
        ann_file=data_root + 'annotations/instances_val2017.json',
        img_prefix=data_root + 'val2017/',
        pipeline=test_pipeline),
    test=dict(
        type=dataset_type,
        ann_file=data_root + 'annotations/instances_val2017.json',
        img_prefix=data_root + 'val2017/',
        pipeline=test_pipeline))
# optimizer
optimizer = dict(type='SGD', lr=0.00125, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=dict(max_norm=35, norm_type=2))
# learning policy
lr_config = dict(
    policy='step',
    warmup='linear',
    warmup_iters=500,
    warmup_ratio=1.0 / 3,
    step=[27, 33])
checkpoint_config = dict(interval=1)
# yapf:disable
log_config = dict(
    interval=10,
    hooks=[
        dict(type='TextLoggerHook'),
        # dict(type='TensorboardLoggerHook')
    ])
# yapf:enable
# runtime settings
total_epochs = 36
device_ids = range(8)
dist_params = dict(backend='nccl')
log_level = 'INFO'
work_dir = './work_dirs/solo_r101_3x'
load_from = None
resume_from = None
workflow = [('train', 1)]

```
only log.json saved into work_dir, what did I miss?

**Comment by jinfagang** [comment]
No log output on terminal, not even mmdetection version info

**Comment by Bai426** [comment]
I have the same problem when I train my model, it seems not to be trained because it took onel a few minutes for training, while several days ago I need 10 hours to train.
2020-04-14 15:23:10,225 - mmdet - INFO - load model from: torchvision://resnet101
2020-04-14 15:23:10,426 - mmdet - WARNING - The model and loaded state dict do not match exactly

unexpected key in source state_dict: fc.weight, fc.bias

loading annotations into memory...
Done (t=0.05s)
creating index...
index created!
2020-04-14 15:23:11,914 - mmdet - INFO - Start running, host: bys2058@bys2058, work_dir: /home/bys2058/work_dirs/mask_rcnn_r101_fpn_1x
2020-04-14 15:23:11,914 - mmdet - INFO - workflow: [('train', 1)], max: 12 epochs

Process finished with exit code 0

**Comment by jinfagang** [comment]
what do u mean? It actually training since my log json generated and I can see it log every 10 interval just like I configed it.

But I can't see any outputs in terminal.

**Comment by Bai426** [comment]
Perhaps your problem is not exactly similar to me.  I cannot train my model and don't know why.

**Comment by hellock** [comment]
That's weird. Are you just using the official solov2 repo?

**Comment by jinfagang** [comment]
Yes. Doesn't change anything, it trained but no log on terminal, however generated into json file.

**Comment by hellock** [comment]
Were you running this repo? https://github.com/Epiphqny/SOLOv2
I just tried it and the logs were printed in the terminal.

**Comment by Bai426** [comment]
Hi, @
hellock<https://github.com/hellock>
Do you know the reason why I cannot train my model any more, even though I reinstalled MMdetection again. it shows like this:
2020-04-20 10:31:18,552 - mmdet - INFO - load model from: torchvision://resnet50
2020-04-20 10:31:18,655 - mmdet - WARNING - The model and loaded state dict do not match exactly

unexpected key in source state_dict: fc.weight, fc.bias

loading annotations into memory...
Done (t=0.03s)
creating index...
index created!
2020-04-20 10:31:20,197 - mmdet - INFO - Start running, host: bys2058@bys2058, work_dir: /home/bys2058/work_dirs/cascade_mask_rcnn_r50_fpn_1xa
2020-04-20 10:31:20,197 - mmdet - INFO - workflow: [('train', 1)], max: 10 epochs

Process finished with exit code 0

Just a few days ago, I can train most models in MMdetection.
Thanks!
________________________________

**Comment by BeBeauty** [comment]
> Perhaps your problem is not exactly similar to me. I cannot train my model and don't know why.

Facing the same issue. Can anyone give some solution?

**Comment by littleTaodog** [comment]
I get same problem, there` no loss output during training.

**Comment by Bai426** [comment]
@littleTaodog, yes, I once faced the same problem as you said. I found it is a problem when 'enumerate' is used. You can try #2499, which title is Error in enumerate(data_loader) #2499.

**Comment by Sakura1221** [comment]
> Yes. Doesn't change anything, it trained but no log on terminal, however generated into json file.

I met the same problem. Have you solve it? @jinfagang

**Comment by YoelShoshan** [comment]
I experienced no log during training recently, I don't know if it's your case as well but sharing anyway.

In my case, for reasons irrelevant to the explanation, each epoch was made of a small amount of batches, (less than 50), 
and since log_config's interval=... defaults to a larger number, logging was never triggered.

https://github.com/open-mmlab/mmdetection/blob/master/configs/_base_/default_runtime.py#L4

The solution for me was simply to reduce the interval value (either that or modify the epoch definition)

**Comment by nizhenliang** [comment]
I met the same problem. Can someone help to solve the problem？

**Comment by Roger-F** [comment]
Same problem here. Anyone?
